scripts/azureml-assets/azureml/assets/release/group_test.py

```diff
@@ -40,42 +80,91 @@
     submitted_job_list = []
     succeeded_jobs = []
     failed_jobs = []
+    test_coverage = defaultdict(list)
+    covered_assets = []
+
     if group_pre:
         check_call(f"python {group_pre}", env=my_env, shell=True)
 
+    pytest_jobs = defaultdict(list)  # pytest job path -> assets coverage dict
     with open(tests_dir / TEST_YML) as fp:
         data = yaml.load(fp, Loader=yaml.FullLoader)
         for job, job_data in data[test_group]['jobs'].items():
-            if 'pre' in job_data:
-                print(f"Running pre script for {job}")
-                proc = check_call(f"python3 {tests_dir / job_data['pre']}", env=my_env, shell=True)
-            print(f'Loading test job {job}')
-            test_job = azure.ai.ml.load_job(tests_dir / job_data['job'])
-            print(test_job)
-            print(f'Running test job {job}')
-            test_job = ml_client.jobs.create_or_update(test_job)
-            print(f'Submitted test job {job}')
-            print(f"Job id: {test_job.id}")
-            submitted_job_list.append(test_job)
+            if 'pytest_job' in job_data:
+                pytest_jobs[tests_dir / job_data['pytest_job']] = job_data["assets"]
+            else:
+                if 'pre' in job_data:
+                    logger.print(f"Running pre script for {job}")
+                    proc = check_call(f"python3 {tests_dir / job_data['pre']}", env=my_env, shell=True)
+                print(f'Loading test job {job}')
+                try:
+                    test_job = azure.ai.ml.load_job(tests_dir / job_data['job'])
+                    logger.print(f'Running test job {job}')
+                    test_job = ml_client.jobs.create_or_update(test_job)
+                except Exception as ex:
+                    logger.log_warning(
+                        f"catch error submitting {job_data['job']} with exception {ex}")
+                    failed_jobs.append(job)
+                    continue
+                test_coverage[test_job] = job_data.get("assets", [])
+                logger.print(f'Submitted test job {job}')
+                logger.print(f"Job id: {test_job.id}")
+                submitted_job_list.append(test_job)
+
+    # Run pytest jobs
+    submitted_pytest_jobs = run_pytest_jobs(pytest_jobs, my_env)
 
+    # Process pytest results
+    # while submitted_pytest_jobs:
+    #    for job in submitted_pytest_jobs.copy().keys():
+    #        if job.done():
+    #            if job.result() == 0:
+    #                succeeded_jobs.append(job)
+    #                covered_assets.extend(submitted_pytest_jobs[job])
+    #            else:
+    #                failed_jobs.append(job)
+    #            del submitted_pytest_jobs[job]
+
+    for job in submitted_pytest_jobs.keys():
+        if job.result() == 0:
+            succeeded_jobs.append(job)
+            covered_assets.extend(submitted_pytest_jobs[job])
+        else:
+            failed_jobs.append(job)
```

Comment:
No need to do a secondary dict lookup:
```suggestion
    for job, assets in submitted_pytest_jobs.items():
        if job.result() == 0:
            succeeded_jobs.append(job)
            covered_assets.extend(assets)
        else:
            failed_jobs.append(job)
```